docs/wfcython.md

```diff
@@ -0,0 +1,612 @@
+
+# A Wright-Fisher simulation implemented in C++ via Cython.
+
+This tutorial implements a Wright-Fisher simulation with mutation and recombination using [Cython](http://www.cython.org).  Cython is two things:
+
+* A grammer/dialect of Python that allows static typing Python and of C/C++ types.
+* A static compiler to turn the Cython grammer in to C or C++ code to compile into a Python extension module.
+
+Cython has a learning curve of its own. A lot of what is shown below reflects best practices.  For those, we refer you to the [Cython documentation](https://cython.readthedocs.io/en/latest/).
+
+Here, we avoid all use of [numpy](http://www.numpy.org) until we have to talk to [msprime](http://msprime.readthedocs.io).  We replace all numpy functionality with the equivalent routines from the C++11 standard library or the excellent GNU Scientific Library, or [GSL](https://www.gnu.org/software/gsl/doc/html/index.html). Yes, numpy is fast!  Numpy is written in C!  But, numpy has to talk back and forth to Python, meaning we can out-perform it by writing routines that execute completely on the C/C++ side.
```

Comment:
Replace what? I think we should refer to the other tutorial a bit here maybe. In the other tutorial we do X, but here we do [something else]. This numpy comment doesn't make much sense without the context.